repo/common/common.go

```diff
@@ -2,26 +2,50 @@ package common
 
 import (
 	"fmt"
+	"regexp"
 	"strings"
 )
 
+// Find dynamic map key names passed  as Parent["foo"] notation
+var bracketsRe = regexp.MustCompile(`\[([^\[\]]*)\]`)
+
+// Normalization for supporting arbitrary dynamic keys with dots:
+// Gateway.PublicGateways["gw.example.com"].UseSubdomains
+// Pinning.RemoteServices["pins.example.org"].Policies.MFS.Enable
```

Comment:
Is this notation standard/used in other places or are we just making this up? Either way we'd need to document this in the `config` command.

Reply:
This is how keys in JSON objects are addressed on the web, and since we use JSON for config anyway, means we don't invent anything new, but follow existing convention. 

I've added note about this notation to `ipfs config --help` in 0a2336ea1

Reply:
The return value is not documented accurately on a public method. What about dynamicKeys?